### PR TITLE
AI Tutor: style compilation and validation support as chat messages

### PR DESCRIPTION
**Pull request by Erin007**
Another UI iteration! I consolidated the `generalChatTutor`, `compilationTutor` and `validationTutor` into a single component, `AITutor` and now display compilation and validation support as chat messages. 

https://github.com/code-dot-org/code-dot-org/assets/12300669/f988168e-ec32-4884-b182-ae79c09885bc

Along the way, I also consolidated the `askAITutor` and `submitChat` thunks since they basically did the same thing when all types of support were converted to chat messages.  Additionally, I moved the select tutor type buttons to below the chat message workspace. 

### Links
[CT-345](https://codedotorg.atlassian.net/browse/CT-345)
[AI Tutor Product Spec](https://docs.google.com/document/d/1rEWpIA6zXzevldiyi6DH-Pt4UU7uqPlWjTnUbz9uVLE/edit#heading=h.qi8xxyq72bct)

### Deployment strategy
Everything is still behind all the same AI Tutor permissions and experiments. 

### PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/aiTutor/redux/aiTutorRedux.ts [hunk 1]
```diff
@@ -87,7 +87,18 @@ export const askAITutor = createAsyncThunk(
...
+    if (chatApiResponse.assistantResponse) {
+      const assistantChatMessage: ChatCompletionMessage = {
+        id: newMessageId,
+        role: Role.ASSISTANT,
+        status: Status.OK,
+        chatMessageText: chatApiResponse.assistantResponse,
+      };
+      thunkAPI.dispatch(addChatMessage(assistantChatMessage));
+    }
+
```
Comment (Erin007) [inline apps/src/aiTutor/redux/aiTutorRedux.ts]: Can/should we consolidate `askAITutor` and `submitChatMessage` thunks?

File: apps/src/code-studio/components/aiTutor/chatWorkspace.tsx [hunk 2]
```diff
@@ -28,16 +35,18 @@ const ChatWorkspace: React.FunctionComponent = () => {
...
         ))}
         {showWaitingAnimation()}
       </div>
-      <div id="chat-workspace-editor" className={style.userChatMessageEditor}>
-        <UserChatMessageEditor />
-      </div>
+      {generalChat && (
+        <div id="chat-workspace-editor" className={style.userChatMessageEditor}>
+          <UserChatMessageEditor />
+        </div>
```
Comment (Erin007) [inline apps/src/code-studio/components/aiTutor/chatWorkspace.tsx]: Do we not hide the whole `UserChatMessageEditor` but instead hide the text input box?

File: apps/src/code-studio/components/aiTutor/selectCompilationButton.tsx [hunk 3]
```diff
@@ -1,14 +1,60 @@
...
+    if (hasRunOrTestedCode && hasCompilationError) {
+      dispatch(
+        addChatMessage({
+          id: 2,
+          role: Role.ASSISTANT,
+          status: Status.OK,
+          chatMessageText:
+            'Ah! You do have an error. Submit your code, and I will try to help.',
+        })
+      );
```
Comment (Erin007) [inline apps/src/code-studio/components/aiTutor/selectCompilationButton.tsx]: This shenanigans works on the first render, but goes sideways fast when we have repeat requests to AI. We'll want to suss out when to add which chat message and handle the chat message id incrementation.

File: apps/src/aiTutor/redux/aiTutorRedux.ts (hunk 1 above)
Comment (Erin007) [reply apps/src/aiTutor/redux/aiTutorRedux.ts]: yep

**Comment by Erin007** [comment]
Reviewers: This PR would be a good one to pull and play around with locally, please! There are a TON of different iterations of using different tutor in different orders and different states the user's tests and code could be in. I've tested locally a fair bit, but see if you can break it, please!

File: apps/src/code-studio/components/aiTutor/aiTutor.tsx [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+  const isRunning = useAppSelector(state => state.javalab.isRunning);
+  const hasRunOrTestedCode = useAppSelector(
+    state => state.javalab.hasRunOrTestedCode
+  );
+  const validationPassed = useAppSelector(
+    state => state.javalab.validationPassed
+  );
+
+  const setCompilationChatMessages = useCallback(() => {
+    const runCode = {
```
Comment (ebeastlake) [inline apps/src/code-studio/components/aiTutor/aiTutor.tsx]: Is there an AI Tutor constants file yet? If so, these could go there.

**Review by ebeastlake — APPROVED** [review]
Nice work!

File: apps/src/code-studio/components/aiTutor/aiTutor.tsx (hunk 4 above)
Comment (Erin007) [reply apps/src/code-studio/components/aiTutor/aiTutor.tsx]: yep, good idea. I'll move them over to the constants file

**Pull request merged**